src/whylogs/app/logger.py

```diff
@@ -34,15 +37,20 @@ def __init__(self,
         dataset_name: str,
         dataset_timestamp: Optional[datetime.datetime] = None,
         session_timestamp: Optional[datetime.datetime] = None,
-        tags: typing.Dict[str, str] = None,
-        metadata: typing.Dict[str, str] = None,
+        tags: Dict[str, str] = {},
+        metadata: Dict[str, str] = None,
         writers = List[Writer],
         verbose: bool = False, 
         with_rotation_time: Optional[str] = None, 
         cache: int =1,
+        segments: Optional[Union[List[List[Dict]], List[str]]] = None,
```

Comment:
this signature is very nested - I'd rather keep type hierarchy lighweight. For the `key, value` case, can we just stick to defining an object instead?

Reply:
Agree. it is essentially `SegmentTag`.

Reply:
done